File: solutions/zig/rna-transcription/1/rna_transcription.zig
const std = @import("std");
const mem = std.mem;

pub fn toRna(allocator: mem.Allocator, dna: []const u8) mem.Allocator.Error![]const u8 {
    const rna = try allocator.alloc(u8, dna.len);
    errdefer allocator.free(rna);

    for (dna, 0..) |nucleotide, i| {
        rna[i] = switch (nucleotide) {
            'G' => 'C',
            'C' => 'G',
            'T' => 'A',
            'A' => 'U',
            else => unreachable,
        };
    }

    return rna;
}

File: solutions/php/rna-transcription/1/RnaTranscription.php
<?php
declare(strict_types=1);

const DNA_TO_RNA = ['G' => 'C', 'C' => 'G', 'T' => 'A', 'A' => 'U'];

function toRna(string $dna): string
{
    $rna = '';
    foreach (str_split($dna) as $nucleotide) {
        $rna .= DNA_TO_RNA[$nucleotide];
    }
    return $rna;
}
